src/interceptors/XMLHttpRequest/XMLHttpRequestProxy.ts

```diff
@@ -99,31 +99,30 @@ export function createXMLHttpRequestProxy({
             return
           }
 
-          // Treat unhandled exceptions in the "request" listener
-          // as 500 server errors.
-          xhrRequestController.respondWith(
-            new Response(
-              JSON.stringify({
-                name: resolverResult.error.name,
-                message: resolverResult.error.message,
-                stack: resolverResult.error.stack,
-              }),
-              {
-                status: 500,
-                statusText: 'Unhandled Exception',
-                headers: {
-                  'Content-Type': 'application/json',
-                },
-              }
+          if (resolverResult.error instanceof Error) {
+            // Treat unhandled exceptions in the "request" listener
+            // as 500 server errors.
+            xhrRequestController.respondWith(
+              new Response(
+                JSON.stringify({
+                  name: resolverResult.error.name,
+                  message: resolverResult.error.message,
+                  stack: resolverResult.error.stack,
+                }),
+                {
+                  status: 500,
+                  statusText: 'Unhandled Exception',
+                  headers: {
+                    'Content-Type': 'application/json',
+                  },
+                }
+              )
             )
-          )
+            return
+          }
 
-          /**
-           * @todo Consider forwarding this error to the stderr as well
-           * since not all consumers are expecting to handle errors.
-           * If they don't, this error will be swallowed.
-           */
-          // xhrRequestController.errorWith(resolverResult.error)
+          // Otherwise, error the request with the thrown data.
+          xhrRequestController.errorWith(resolverResult.error)
```

Comment:
The argument for translating any other (non-error) data to request errors is a bit weaker. There should be a way to trigger request errors.

Now that I think about it, we do have that way! It's this:

```js
request.respondWith(Response.error())
```

This translates to a request error across different clients. **I think we need to remove non-error handling changes and treat non-errors as 500 responses anyway.**